### ISSUE TITLE
`PlayGamesUserProfile.Equals(object obj)` implemented incorrectly.

### ISSUE THREAD

**Issue by Qbit86** [playgameservices/play-games-plugin-for-unity#1180]
This condition is meaningless:
```
if (!typeof(object).IsSubclassOf(typeof(PlayGamesUserProfile)))
//             ^^^
{
    return false;
}

...
```
It should be something like that:
```
PlayGamesUserProfile other = obj as PlayGamesUserProfile;
if (other == null)
{
    return false;
}

return StringComparer.Ordinal.Equals(mPlayerId, other.mPlayerId);
```

**Comment by claywilkinson** [comment]
Thanks for the feedback! Please feel free to submit a pull request to improve the code when you come across opportunities like this.

**Comment by claywilkinson** [comment]
I am adding this to an upcoming update.

**Comment by claywilkinson** [comment]
Fixed in 0.9.34
